### PR TITLE
feat[auth]: add login with google

### PR DESCRIPTION
**Pull request by thormengkheang**
I introduced 3 new environment variables

`BASE_URL=http://localhost:3000` it's for google auth client redirect
`GOOGLE_CLIENT_ID=`
`GOOGLE_CLIENT_SECRET=`

I have also added AuthContoller [33-google-login/src/controllers/auth.ts](https://github.com/invisal/libsql-studio/blob/33-google-login/src/controllers/auth.ts) for easier to integrate with new provider.

To add login in with google, first need to create a new project at https://console.cloud.google.com/

Then create a consent screen at https://console.cloud.google.com/apis/credentials/consent with the scope below

Screenshot 2024-03-20 at 4 33 56 in the afternoon

Then need to create 2 credentials OAuth client ID one for development and another one for production

for development need to config like below

Screenshot 2024-03-20 at 4 33 22 in the afternoon

### CODE REVIEW

File: src/lib/auth.ts [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 import { env } from "@/env";
 import { cache } from "react";
 import { cookies, headers } from "next/headers";
 
+export const PROVIDER = {
+  GITHUB: "GITHUB",
+  GOOGLE: "GOOGLE",
+} as const;
+
+type ObjectValues<T> = T[keyof T];
```
Comment (rin-yato) [inline src/lib/auth.ts:14]: Maybe we can put this in another file?
e.g. `@/lib/type-util`
This seems to be reusable later

File: src/app/login/page.tsx [hunk 2]
```diff
@@ -10,10 +10,13 @@ export default function LoginPage() {
...
             <h2 className="text-xl font-bold">Sign In</h2>
           </CardHeader>
-          <CardContent>
+          <CardContent className="grid gap-4">
             <Link href="/login/github">
               <Button>Continue with Github</Button>
             </Link>
+            <Link href="/login/google">
+              <Button>Continue with Google</Button>
+            </Link>
```
Comment (rin-yato) [inline src/app/login/page.tsx:19]: In the case of accessibility, it is better to do:
```suggestion
	    <Button asChild>
              <Link href="/login/google">
                Continue with Google
              </Link>
            </Button>
```

**Review by rin-yato — APPROVED** [review]
Overall looks good to me!🔥

File: src/lib/auth.ts (hunk 1 above)
Comment (thormengkheang) [reply src/lib/auth.ts:14]: yeah we can move it definition file (d.ts) or global but currently only use it here

File: src/app/login/page.tsx (hunk 2 above)
Comment (thormengkheang) [reply src/app/login/page.tsx:19]: I see thanks for the suggestion!

**Review by invisal — APPROVED** [review]
LGTM

**Pull request merged**